src/treemodel/delta/transform.js

```diff
@@ -0,0 +1,490 @@
+/**
+ * @license Copyright (c) 2003-2016, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+'use strict';
+
+import OT from '../operation/transform.js';
+
+import Range from '../range.js';
+
+import AttributeOperation from '../operation/attributeoperation.js';
+
+import Delta from './delta.js';
+import AttributeDelta from './attributedelta.js';
+import InsertDelta from './insertdelta.js';
+import MergeDelta from './mergedelta.js';
+import MoveDelta from './movedelta.js';
+import RemoveDelta from './removedelta.js';
+import SplitDelta from './splitdelta.js';
+import WeakInsertDelta from './weakinsertdelta.js';
+import WrapDelta from './wrapdelta.js';
+import UnwrapDelta from './unwrapdelta.js';
+
+import arrayUtils from '../../lib/lodash/array.js';
+import utils from '../../utils.js';
+
+/**
+ * @namespace core.treeModel.delta.transform
+ */
+
+/**
+ * When it comes to delta transformations, delta has their priorities. Priorities mean which delta should be a
+ * stronger one when conflicting situation arises. These depend only on delta type. The higher the priority, the stronger
+ * the delta is. For example, then MoveDelta tries to move some nodes and SplitDelta tries to move the some of those
+ * nodes to a new parent, MoveDelta should be treated as stronger and the result of MoveDelta should overwrite SplitDelta.
+ * When deltas have same priorities, `isStrong` parameter is used to decide which one is stronger. This parameter should
+ * be arbitrary set, i.e. depending on the source of the delta (i.e. during collaborative editing deltas coming from
+ * server might be more important than deltas produced on client).
+ *
+ * @external core.treeModel.delta.transform
+ * @member core.treeModel.delta.transform.priorities
+ * @type {Map}
+ */
+export const priorities = new Map();
+priorities.set( MergeDelta, 0 );
+priorities.set( UnwrapDelta, 1 );
+priorities.set( WrapDelta, 1 );
+priorities.set( SplitDelta, 1 );
+priorities.set( AttributeDelta, 2 );
+priorities.set( InsertDelta, 2 );
+priorities.set( MoveDelta, 2 );
+priorities.set( RemoveDelta, 3 );
```

Comment:
Maybe there priorities should be: 10, 20, 30, in case someone want to put his delta between existing.